### PR TITLE
New Issue templates

### PR DESCRIPTION
**Pull request by publicarray**
GitHub has a new template format and I wanted to try it. You can try it yourself here: https://github.com/publicarray/spksrc/issues/new/choose

sample output of a bug report: https://github.com/publicarray/spksrc/issues/31

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -1,7 +1,23 @@
...
+## Description
+
+Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context.
+
+Fixes # Optionally, add links to existing issues or other PR's
+
+## Checklist
 
-### Checklist
-- [ ] Build rule `all-supported` completed successfully
```
Comment (hgy59) [inline .github/PULL_REQUEST_TEMPLATE.md:5]: I would like to keep the Build ... completed... Check to motivate contributors to build all-supported before creating a PR.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -1,7 +1,23 @@
...
+
+
+### Type of change
+
+Please use any relavent tags.
+
+- [ ] Bug fix
+- [ ] New Package
+- [ ] Package update
+- [ ] Includes framework changes
```
Comment (hgy59) [inline .github/PULL_REQUEST_TEMPLATE.md:22]: When ever possible I would create a different PR template for framework changes.
- we should create dedicated PRs for framework changes (/mk/*) as those do not trigger package builds and require extended build tests
- the checklist for framework changes would look different...
-

**Comment by hgy59** [comment]
Just an idea: can we provide a shell script (or spk package) that asks for a package and creates an archive file (zip/tar.gz) that containes the relevant information for a bug-report?

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (publicarray) [reply .github/PULL_REQUEST_TEMPLATE.md:5]: I see your point and added it back, but I haven't done it myself for quite a while. The CI does this mostly for us anyway.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (publicarray) [reply .github/PULL_REQUEST_TEMPLATE.md]: [GitHub](https://github.community/t/is-there-a-pull-request-template-selector-similar-to-issues/171838) [doesn't have a GUI for it](https://github.community/t/multiple-pull-request-templates/1850/12) right now, but you can use the `?template=` parameter to choose the framework PR template:

https://github.com/publicarray/spksrc/compare/master...issue-template?template=framework.md

**Comment by publicarray** [comment]
Yea I had a similar idea when I wrote the test script which already grabs most of that information: https://github.com/publicarray/spkcli

An easy way to at fill out the form is with URL parameters: 

https://github.com/publicarray/spksrc/issues/new?assignees=&labels=bug&template=bug_report.yml&package=Jellyfin&version=1.0&model=DS%20123&arch=x64&firmware=DSM%206.0

(the `id:` is the URL parameter)

**Comment by publicarray** [comment]
@hgy59 I think this is good to merge but feel free to change it. 

Once in use we might find things where I made wrong assumption. Like maybe log file should be attached as a file?

**Review by hgy59 — APPROVED** [review]
LGTM

**Pull request merged**